addons/l10n_in/models/res_config_settings.py

```diff
@@ -33,3 +53,30 @@ def l10n_in_edi_buy_iap(self):
             'url': self.env["iap.account"].get_credits_url(service_name=IAP_SERVICE_NAME),
             'target': '_new'
         }
+
+    def set_values(self):
+        if self.country_code == 'IN':
+            old_value = str2bool(self.env['ir.config_parameter'].get_param('l10n_in.l10n_in_is_gst_registered'))
+            if old_value != self.l10n_in_is_gst_registered:
```

Comment:
inheriting `set_values` is not a good idea because it will trigger every time whenever anyone tries to save the res config setting.

For the better suggestion please create the fields in `res.company` which will be related to `res.config.settings` can and make sure the field is dependent on company's `vat\GSTIN`.

May be compute won't hurt?